adapters/videoheroes/videoheroes.go

```diff
@@ -0,0 +1,164 @@
+package videoheroes
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb/v16/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+// Builder builds a new instance of the VideoHeroes adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	uri, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: uri,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+	var videoHeoresExt *openrtb_ext.ExtImpVideoHeroes
+	var err error
+
+	for i, imp := range request.Imp {
+		videoHeoresExt, err = a.getImpressionExt(&imp)
```

Comment:
Optimization suggestion: instead of calling `getImpressionExt` for every impression, execute it once before the for loop for the firs imp. 
Question: in one request can impressions have different placementIds?

Reply:
I suppose its not required in this case and can removed as well, to use only first index of array. Thanks !